src/task_queue/database_tasks/media_library_cleanup.py

```diff
@@ -0,0 +1,63 @@
+import datetime
+from django.http import HttpResponse
+from _main_.utils.emailer.send_email import send_massenergize_email_with_attachments
+from api.store.common import (
+    find_duplicate_items,
+    get_admins_of_communities,
+    get_duplicate_count,
+    remove_duplicates_and_attach_relations,
+    summarize_duplicates_into_csv,
+)
+from api.utils.constants import MEDIA_LIBRARY_CLEANUP_TEMPLATE
+from database.models import FeatureFlag
+from task_queue.models import Task
+
+
+REMOVE_DUPLICATE_IMAGE_FLAG_KEY = "remove-duplicate-images-feature-flag"
+
+
+def remove_duplicate_images():
+    """
+    This checks all media on the platform and removes all duplicates.
+    Its based on the "Remove Duplicate Images" feature flag. For communities that are subscribed
+    to the flag, duplicates will be removed from their libraries.
+    """
+    try: 
+        flag = FeatureFlag.objects.filter(key=REMOVE_DUPLICATE_IMAGE_FLAG_KEY).first()
+        communities = flag.enabled_communities()
+        task = Task.objects.filter(name = "Media Library Cleanup Routine").first()
```

Comment:
This shoudl be filtered by the process name "Remove Duplicate Images" since the task creator is free to name the task whatever they want.  But even better, you can pass the task into the routine, see Tahiru's examples

Reply:
I guess I will call him 🤣 🤣 🤣  cos I quickly looked around to see if there was a chance that I could access the task directly in there and didnt  find any example.